src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -266,6 +267,7 @@ entails SolverOptions{..} constraint context hints =
 
             canBeGeneralized :: Type -> Bool
             canBeGeneralized TUnknown{} = True
+            canBeGeneralized (KindedType t _) = canBeGeneralized t
```

Comment:
Okay it does work, but only if I make it a separate declaration and not in a let. :smile: